google/cloud/forseti/services/scanner/dao.py

```diff
@@ -190,3 +199,34 @@ def map_by_resource(violation_rows):
             v_by_type[v_resource].append(v_data)
 
     return dict(v_by_type)
+
+def _create_violation_hash(violation, inventory_index_id,
+                           algorithm=FALLBACK_HASH_ALGORITHM):
+    """Create a hash of violation data using the requested algorithm.
+
+    Args:
+        violation (dict): A violation.
+        inventory_index_id (str): Id of the inventory index.
+        algorithm (str): The algorithm used to create the hash.
+
+    Returns:
+        str: The resulting hex digest or '' if we can't successfully create
+        a hash.
+    """
+    try:
+        violation_hash = hashlib.new(algorithm)
+    except ValueError as e:
+        LOGGER.error('Unable to use the specified hash algorithm: %s',
```

Comment:
It's hard to see where someone can specify a different algorithm, without changing the code.
So, it seems this hashing would run twice on error, and fail twice.

Reply:
The `FALLBACK_HASH_ALGORITHM` is known to work and will be used if the caller specifies an algorithm that fails.

Reply:
Where would a caller be expected to pass an algorithm in? This is a private method inside of the module, so it should not have any external callers.